### ISSUE TITLE
Getting error while installing oracle - 12.2.0.1

### ISSUE THREAD

**Issue by hiteshtalhilyani** [biemond/biemond-oradb#187]
Hi 
 I am getting error while installing the oracle "12.2.0.1" version, is it the bug in the module for this version or it can be solved. if yes, can you please guide me ?
```
oradb::installdb{ '12.2.0.1_Linux-x86-64':
        version                   => '12.2.0.1',

Notice: /Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns:  - Action: User Action Not Available
Error: '/bin/sh -c 'unset DISPLAY;/oraapp/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /oraapp/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'' returned 254 instead of one of [6,0]
**Error: /Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns: change from notrun to 6 0 failed: '/bin/sh -c 'unset DISPLAY;/oraapp/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /oraapp/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'' returned 254 instead of one of [6,0]**
Notice: /Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[run root.sh script 12.2.0.1_Linux-x86-64]: Dependency Exec[install oracle database 12.2.0.1_Linux-x86-64] has failures: true
```

Regards 
Hitesh Talhilyani

**Comment by biemond** [comment]
Hi, 

can you check the install log on the VM or show the real output plus make sure you install all the yum dependencies and set the kernel limits.

thanks Edwin

**Comment by hiteshtalhilyani** [comment]
Hi, 
I have installed all dependent packages and kernel parameters as below:

```
[root@razordhcp puppet]# cat package.pp
$install = [ 'binutils.x86_64', 'compat-libstdc++-33.x86_64', 'glibc.x86_64','ksh.x86_64','libaio.x86_64',
             'libgcc.x86_64', 'libstdc++.x86_64', 'make.x86_64','compat-libcap1.x86_64', 'gcc.x86_64',
             'gcc-c++.x86_64','glibc-devel.x86_64','libaio-devel.x86_64','libstdc++-devel.x86_64',
             'sysstat.x86_64','unixODBC-devel','glibc.i686','libXext.x86_64','libXtst.x86_64']
 package { $install:
  ensure  => present,
}
[root@razordhcp puppet]# puppet apply package.pp
Notice: Compiled catalog for razordhcp.emaratech.ae in environment production in 0.21 seconds
Notice: Applied catalog in 2.69 seconds
[root@razordhcp puppet]#
[root@razordhcp puppet]#
[root@razordhcp puppet]# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.3 (Maipo)

[root@razordhcp puppet]# sysctl -p
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
net.ipv6.conf.lo.disable_ipv6 = 1
kernel.msgmnb = 65536
kernel.msgmax = 65536
kernel.shmmax = 2588483584
kernel.shmall = 2097152
fs.file-max = 6815744
net.ipv4.tcp_keepalive_time = 1800
net.ipv4.tcp_keepalive_intvl = 30
net.ipv4.tcp_keepalive_probes = 5
net.ipv4.tcp_fin_timeout = 30
kernel.shmmni = 4096
fs.aio-max-nr = 1048576
kernel.sem = 250 32000 100 128
net.ipv4.ip_local_port_range = 9000 65500
net.core.rmem_default = 262144
net.core.rmem_max = 4194304
net.core.wmem_default = 262144
net.core.wmem_max = 1048576

And I am getting below error during in VM install log:

Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Starting Oracle Universal Installer...
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Checking Temp space: must be greater than 500 MB.   Actual 7125 MB    Passed
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Checking swap space: must be greater than 150 MB.   Actual 10575 MB    Passed
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Preparing to launch Oracle Universal Installer from /tmp/OraInstall2017-08-30_06-08-34PM. Please wait ...[FATAL] [INS-30131] Initial setup required for the execution of installer validations failed.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    CAUSE: Failed to access the temporary location.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    ACTION: Ensure that the current user has required permissions to access the temporary location.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) *ADDITIONAL INFORMATION:*
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - PRVG-1901 : failed to setup CVU remote execution framework directory "/tmp/CVU_12.2.0.1.0_oracle/" on nodes "razordhcp"
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Cause:  An operation requiring remote execution could not complete because the attempt to set up the Cluster Verification Utility remote execution framework failed on the indicated nodes at the indicated directory location because the CVU remote execution framework version did not match the CVU java verification framework version. The accompanying message provides detailed failure information.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Action:  Ensure that the directory indicated exists or can be created and the user executing the checks has sufficient permission to overwrite the contents of this directory. Also review the accompanying error messages and respond to them.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Summary of the failed nodes
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) razordhcp
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Version of exectask could not be retrieved from node "razordhcp"
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Cause: Cause Of Problem Not Available
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Action: User Action Not Available
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Version of exectask could not be retrieved from node "razordhcp"
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Cause: Cause Of Problem Not Available
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Action: User Action Not Available
Aug 30 18:09:45 razordhcp puppet-agent[16121]: '/bin/sh -c 'unset DISPLAY;/oraapp/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /oraapp/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'' returned 254 instead of one of [6,0]
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) change from notrun to 6 0 failed: '/bin/sh -c 'unset DISPLAY;/oraapp/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /oraapp/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'' returned 254 instead of one of [6,0]
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[run root.sh script 12.2.0.1_Linux-x86-64]) Dependency Exec[install oracle database 12.2.0.1_Linux-x86-64] has failures: true
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[run root.sh script 12.2.0.1_Linux-x86-64]) Skipping because of failed dependencies
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/oraapp/product/12.2/db]) Dependency Exec[install oracle database 12.2.0.1_Linux-x86-64] has failures: true
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/oraapp/product/12.2/db]) Skipping because of failed dependencies
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[remove oracle db extract folder 12.2.0.1_Linux-x86-64]) Dependency Exec[install oracle database 12.2.0.1_Linux-x86-64] has failures: true
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[remove oracle db extract folder 12.2.0.1_Linux-x86-64]) Skipping because of failed dependencies
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[remove oracle db file1 V839960-01.zip 12.2.0.1_Linux-x86-64]) Dependency Exec[install oracle database 12.2.0.1_Linux-x86-64] has failures: true
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[remove oracle db file1 V839960-01.zip 12.2.0.1_Linux-x86-64]) Skipping because of failed dependencies
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[remove oracle db file2  12.2.0.1_Linux-x86-64]) Dependency Exec[install oracle database 12.2.0.1_Linux-x86-64] has failures: true
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[remove oracle db file2  12.2.0.1_Linux-x86-64]) Skipping because of failed dependencies
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (Oradb::Installdb[12.2.0.1_Linux-x86-64]) Unscheduling all events on Oradb::Installdb[12.2.0.1_Linux-x86-64]
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (Class[Puppetskills]) Unscheduling all events on Class[Puppetskills]
Aug 30 18:09:59 razordhcp puppet-agent[16121]: (Stage[main]) Unscheduling all events on Stage[main]
Aug 30 18:10:00 razordhcp puppet-agent[16121]: Applied catalog in 226.26 seconds
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=SYSCALL msg=audit(1504102202.108:829): arch=c000003e syscall=2 success=yes exit=3 a0=7f159646e440 a1=2 a2=180 a3=2 items=1 ppid=1174 pid=16838 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="crond" exe="/usr/sbin/crond" key="logins"
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=CWD msg=audit(1504102202.108:829):  cwd="/"
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=PATH msg=audit(1504102202.108:829): item=0 name="/var/run/faillock/root" inode=23329 dev=00:12 mode=0100600 ouid=0 ogid=0 rdev=00:00 objtype=NORMAL
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=EOE msg=audit(1504102202.108:829):
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=USER_ACCT msg=audit(1504102202.116:830): pid=16838 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:accounting grantors=pam_access,pam_faillock,pam_unix,pam_localuser acct="root" exe="/usr/sbin/crond" hostname=? addr=? terminal=cron res=success'
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=CRED_ACQ msg=audit(1504102202.116:831): pid=16838 uid=0 auid=4294967295 ses=4294967295 msg='op=PAM:setcred grantors=pam_env,pam_faillock,pam_unix acct="root" exe="/usr/sbin/crond" hostname=? addr=? terminal=cron res=success'
Aug 30 18:10:02 razordhcp audispd: node=razordhcp.emaratech.ae type=LOGIN msg=audit(1504102202.117:832): pid=16838 uid=0 old-auid=4294967295 auid=0 old-ses=4294967295 ses=48 res=1
Aug 30 18:10:02 razordhcp systemd: Started Session 48 of user root.
```

**Comment by hiteshtalhilyani** [comment]
Hi, 

I have noticed one more thing when I am executing the command manually through oracle user, the installation is working fine.
```
[oracle@razordhcp ~]$ /bin/sh -c 'unset DISPLAY;/oraapp/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /oraapp/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'
Starting Oracle Universal Installer...

Checking Temp space: must be greater than 500 MB.   Actual 5577 MB    Passed
Checking swap space: must be greater than 150 MB.   Actual 10568 MB    Passed
Preparing to launch Oracle Universal Installer from /tmp/OraInstall2017-08-30_06-56-22PM. Please wait ...You can find the log of this install session at:
 /oraInventory/logs/installActions2017-08-30_06-56-22PM.log
```

**Comment by hiteshtalhilyani** [comment]
```
[oracle@razordhcp ~]$ /bin/sh -c 'unset DISPLAY;/oraapp/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /oraapp/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'
Starting Oracle Universal Installer...

Checking Temp space: must be greater than 500 MB.   Actual 5577 MB    Passed
Checking swap space: must be greater than 150 MB.   Actual 10568 MB    Passed
Preparing to launch Oracle Universal Installer from /tmp/OraInstall2017-08-30_06-56-22PM. Please wait ...You can find the log of this install session at:
 /oraInventory/logs/installActions2017-08-30_06-56-22PM.log
The installation of Oracle Database 12c was successful.
Please check '/oraInventory/logs/silentInstall2017-08-30_06-56-22PM.log' for more details.

As a root user, execute the following script(s):
        1. /oraapp/product/12.2/db/root.sh



Successfully Setup Software.
```

**Comment by biemond** [comment]
It can be of this.

```
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Checking Temp space: must be greater than 500 MB.   Actual 7125 MB    Passed
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Checking swap space: must be greater than 150 MB.   Actual 10575 MB    Passed
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Preparing to launch Oracle Universal Installer from /tmp/OraInstall2017-08-30_06-08-34PM. Please wait ...[FATAL] [INS-30131] Initial setup required for the execution of installer validations failed.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    CAUSE: Failed to access the temporary location.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    ACTION: Ensure that the current user has required permissions to access the temporary location.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) *ADDITIONAL INFORMATION:*
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - PRVG-1901 : failed to setup CVU remote execution framework directory "/tmp/CVU_12.2.0.1.0_oracle/" on nodes "razordhcp"
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Cause:  An operation requiring remote execution could not complete because the attempt to set up the Cluster Verification Utility remote execution framework failed on the indicated nodes at the indicated directory location because the CVU remote execution framework version did not match the CVU java verification framework version. The accompanying message provides detailed failure information.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Action:  Ensure that the directory indicated exists or can be created and the user executing the checks has sufficient permission to overwrite the contents of this directory. Also review the accompanying error messages and respond to them.
Aug 30 18:09:45 razordhcp puppet-agent[16121]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Summary of the failed nodes
```

So it can be the oracle user account is not ok when run with su , especially it is missing the temp stuff or it can be it run from a directory where it does not have the permission.

Can you try this , Here I use CentOS , maybe it is a RHEL issue with the oracle profile.
https://github.com/biemond/biemond-oradb-vagrant-12.2-CDB-puppet4_3

**Comment by biemond** [comment]
and maybe it cannot create a dir in the /tmp 
/tmp/CVU_12.2.0.1.0_oracle/

**Comment by hiteshtalhilyani** [comment]
Its creating the directory in /tmp but still giving the error related to /tmp permissions only.

```
[oracle@razordhcp tmp]$ ll
total 4
drwxr-xr-x 2 oracle oinstall 4096 Aug 30 20:22 CVU_12.2.0.1.0_oracle
drwxr-xr-x 2 oracle oinstall    6 Aug 30 20:22 hsperfdata_oracle
[oracle@razordhcp tmp]$ touch test
[oracle@razordhcp tmp]$ ll
total 4
drwxr-xr-x 2 oracle oinstall 4096 Aug 30 20:22 CVU_12.2.0.1.0_oracle
drwxr-xr-x 2 oracle oinstall    6 Aug 30 20:22 hsperfdata_oracle
-rw-r--r-- 1 oracle oinstall    0 Aug 30 20:25 test
---------------------------------------------------------------------------

Aug 30 20:22:28 razordhcp puppet-agent[5109]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86          -64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Preparing to launch Oracle Universal Insta          ller from /tmp/OraInstall2017-08-30_08-21-01PM. Please wait ...[FATAL] [INS-30131] Initial setup required fo          r the execution of installer validations failed.
Aug 30 20:22:28 razordhcp puppet-agent[5109]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86          -64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    CAUSE: Failed to access the temporary l          ocation.
Aug 30 20:22:28 razordhcp puppet-agent[5109]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86          -64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    ACTION: Ensure that the current user ha          s required permissions to access the temporary location.
Aug 30 20:22:28 razordhcp puppet-agent[5109]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86          -64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) *ADDITIONAL INFORMATION:*
Aug 30 20:22:28 razordhcp puppet-agent[5109]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86          -64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - PRVG-1901 : failed to setup CVU remote           execution framework directory "/tmp/CVU_12.2.0.1.0_oracle/" on nodes "razordhcp"
Aug 30 20:22:28 razordhcp puppet-agent[5109]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86          -64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - Cause:  An operation requiring remote e          xecution could not complete because the attempt to set up the Cluster Verification Utility remote execution           framework failed on the indicated nodes at the indicated directory location because the CVU remote execution           framework version did not match the CVU java verification framework version. The accompanying message provi          des detailed failure information.
```

**Comment by biemond** [comment]
maybe we need to set the umask or set a different temp dir in the profile.

**Comment by hiteshtalhilyani** [comment]
I have set the umask value in .bash_profile and changed the /tmp directory too for the oracle user, but while executing it's reverting back the changes.
```
------------------------------------------------------

le[/home/oracle/.bash_profile]/content) --- /home/oracle/.bash_profile#0112017-09-04 11:40:06.258691834 +0400
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content) +++ /tmp/puppet-file20170904-22693-b18ct8#0112017-09-04 11:44:09.155701071 +0400
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content) @@ -19,4 +19,4 @@
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content)
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content)  export TEMP=/tmp
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content)  export TMPDIR=/tmp
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content) -umask 0000
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content) +umask 022
Sep  4 11:44:09 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/File[/home/oracle/.bash_profile]/content) \ No newline at end of file
-------------------------------------------------------------------------
Sep  4 11:44:48 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Preparing to launch Oracle Universal Installer from /tmp/OraInstall2017-09-04_11-44-11AM. Please wait ...[FATAL] [INS-30131] Initial setup required for the execution of installer validations failed.
Sep  4 11:44:48 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    CAUSE: Failed to access the temporary location.
Sep  4 11:44:48 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    ACTION: Ensure that the current user has required permissions to access the temporary location.
Sep  4 11:44:48 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) *ADDITIONAL INFORMATION:*
Sep  4 11:44:48 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) Exception details
Sep  4 11:44:48 oracle_test puppet-agent[22693]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - PRVG-11322 : One or more node names "oracle_test" contain one or more of the following invalid characters "_"
Sep  4 11:44:48 oracle_test puppet-agent[22693]: '/bin/sh -c 'unset DISPLAY;/appvol/V839960-01/database/runInstaller -silent -waitforcompletion -ignoreSysPrereqs -ignorePrereq -responseFile /appvol/db_install_12.2.0.1_12.2.0.1_Linux-x86-64.rsp'' returned 254 instead of one of [6,0]
--------------------------------------------------------------------------------------------------

[root@oracle_test oracle]# cd /tmp
[root@oracle_test tmp]# ll
total 0
drwxr-xr-x. 2 oracle oinstall 6 Sep  4 11:44 hsperfdata_oracle
```

**Comment by biemond** [comment]
can you check this 
PRVG-11322 : One or more node names "oracle_test" contain one or more of the following invalid characters "_"

**Comment by hiteshtalhilyani** [comment]
Now getting this error.
```
Install2017-09-04_05-22-18PM. Please wait ...[FATAL] [INS-30131] Initial setup required for the execution of installer validations failed.
Sep  4 17:23:03 oracletest puppet-agent[15450]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    CAUSE: Failed to access the temporary location.
Sep  4 17:23:03 oracletest puppet-agent[15450]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)    ACTION: Ensure that the current user has required permissions to access the temporary location.
Sep  4 17:23:03 oracletest puppet-agent[15450]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns) *ADDITIONAL INFORMATION:*
Sep  4 17:23:03 oracletest puppet-agent[15450]: (/Stage[main]/Puppetskills/Oradb::Installdb[12.2.0.1_Linux-x86-64]/Exec[install oracle database 12.2.0.1_Linux-x86-64]/returns)  - PRVG-1901 : failed to setup CVU remote execution framework directory "/tmp/CVU_12.2.0.1.0_oracle/" on nodes "oracletest"
---------------------------------------------------------
 In /tmp its writing the files 
[root@oracletest tmp]# ll
total 4
drwxr-xr-x. 3 oracle oinstall 4096 Sep  4 17:23 CVU_12.2.0.1.0_oracle
drwxr-xr-x. 2 oracle oinstall    6 Sep  4 17:23 hsperfdata_oracle
drwxr-x---. 9 oracle oinstall  223 Sep  4 17:23 OraInstall2017-09-04_05-22-18PM
-----------------------------------------------------------------------------------------
Below is the input I am giving

          include oradb
        oradb::installdb{ '12.2.0.1_Linux-x86-64':
        version                   => '12.2.0.1',
        file                      => 'V839960-01',
        database_type             => 'EE',
        oracle_base               => '/appvol/oracle',
        oracle_home               => '/appvol/oracle/product/12.2/db',
        bash_profile              => true,
        user                      => 'oracle',
        group                     => 'dba',
        group_install             => 'oinstall',
        group_oper                => 'oper',
        download_dir              => '/oravol',
        zip_extract               => true,
        puppet_download_mnt_point => '/usr/software',
        }
---------------------------------------------------------------------
```
Can you please tell me how we can specify the two zip file for 11.2.0.4 version

 p13390677_112040_Linux-x86-64_1of7.zip
 p13390677_112040_Linux-x86-64_2of7.zip

**Comment by biemond** [comment]
are you doing 12.2 or 11.2.04? and please look at the examples. I will know it what to do when you specify the version.

**Comment by hiteshtalhilyani** [comment]
I am trying with 12.2 but as it's giving error so want to try with 11.2.04 versions, so in 11.2 i have two zip files.

**Comment by hiteshtalhilyani** [comment]
so biemond, what you would like to suggest for this 12.2 error?? is it possible to fix this temp access error,??

**Comment by biemond** [comment]
do the matching vagrant box and find out what is different?

**Comment by hiteshtalhilyani** [comment]
Issue has resolved now and i am able to install 12.2 on RHEL7 server. Thanks for all your support
